cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -1323,3 +1331,36 @@ fromAlonzoPParams
     , protocolParamMaxCollateralInputs = Just _maxCollateralInputs
     }
 
+-- | Calculate cost of making a UTxO entry for a given 'Value' and
+-- mininimum UTxO value derived from the 'ProtocolParameters'
+calcMinimumDeposit :: ProtocolParameters -> Value -> Lovelace
+calcMinimumDeposit pp v  =
+  case protocolParamMinUTxOValue pp of
+    Nothing ->
+      case protocolParamUTxOCostPerWord pp of
+        Nothing ->
+          error "calcMinimumDeposit: uTxOCostPerWord and minUTxOValue protocol parameters were not found"
+        Just (Lovelace cPerWord) ->
+          -- TODO: Alonzo - ledger should expose a function to calculate the required deposit
```

Comment:
TODO item

Reply:
The size is calculated here:

https://github.com/input-output-hk/cardano-ledger-specs/blob/ecb7275b2de624cede629fec2c4e9237639425e3/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs#L104

and is multiplied by `adaPerUTxOWord` here:

https://github.com/input-output-hk/cardano-ledger-specs/blob/ecb7275b2de624cede629fec2c4e9237639425e3/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs#L394

Reply:
@JaredCorduan That calculates the size of the tx and then the cost (via multiplication by `adaPerUTxOWord`).  We want to only calculate the minimum deposit for a multi asset value.

Reply:
just let me know, we could certainly add something for you. did you want a function that takes a value and a maybe hash? or are you good now?